Frameworks/QuartzCore/CALayer.mm

```diff
@@ -710,26 +585,12 @@ static void doRecursiveAction(CALayer* layer, NSString* actionName) {
     }
 }
 
-- (BOOL)_isPartOfViewHeiarchy {
-    CALayer* curLayer = self;
-
-    while (curLayer != nil) {
-        if (curLayer->priv->isRootLayer) {
-            return TRUE;
-        }
-
-        curLayer = curLayer->priv->superlayer;
-    }
-
-    return FALSE;
-}
-
 /**
  @Status Interoperable
 */
 - (void)addSublayer:(CALayer*)subLayerAddr {
     if (subLayerAddr == self) {
-        assert(0);
+        FAIL_FAST();
```

Comment:
>FAIL_FAST(); [](start = 8, length = 12)

We should really use FAIL_FAST_MSG or accompany this state with a Trace* message.